Extension/src/Utility/System/guards.ts

```diff
@@ -0,0 +1,67 @@
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation. All Rights Reserved.
+ * See 'LICENSE' in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+
+import { Socket } from 'node:net';
+import { isPromise } from 'node:util/types';
+import { AsyncConstructor, Constructor, Primitive } from './types';
+
+// eslint-disable-next-line @typescript-eslint/naming-convention
+export class is {
```

Comment:
Is this going to cause any usability issue in regards to the TypeScript "is" keyword or is the overloaded usage desired? Would it be better as something different like isType?

Reply:
`is` is not a reserved word.

Reply:
Yeah, I didn't say it was a reserved word, I just mentioned that it's a keyword.

Reply:
`is` is not a keyword in JavaScript/TypeScript at all.

Reply:
I'm referring to https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates -- the TypeScript lexical tokenizer reports it as a keyword in regards to lexical colorization. I'm not sure what it's called technically.